src/vcpkg/base/files.cpp

```diff
@@ -3011,6 +3011,27 @@ namespace vcpkg
                 return false;
             }
 
+            if (options == CopyOptions::update_existing)
+            {
+                WIN32_FILE_ATTRIBUTE_DATA attributes_source;
+                if (!GetFileAttributesExW(wide_source.c_str(), GetFileExInfoStandard, &attributes_source))
+                {
+                    ec.assign(GetLastError(), std::system_category());
+                    return false;
+                }
+
+                WIN32_FILE_ATTRIBUTE_DATA attributes_destination;
+                if (GetFileAttributesExW(wide_destination.c_str(), GetFileExInfoStandard, &attributes_destination))
```

Comment:
This call isn't handing errors correctly.

Reply:
Is there a meaningful case where the file exists but we cannot obtain attributes?

Reply:
We don't know that the destination exists here. If the call failed, `attibutes_destination` is bogus, not something that is reasonable to compare against.

Moreover, if it fails with some kind of file I/O error we need to bail and report that, not blindly ignore it.

Reply:
That is, I believe this currently translates "failed to get destination attributes" into "OK, call `CopyFileW(... FALSE)` and destroy the destination" which is not correct.

Reply:
Thanks, I have made the changes